### PR TITLE
added support for .gz files

### PR DESCRIPTION
**Pull request by SultanOrazbayev**

### CODE REVIEW

**Comment by galtay** [comment]
Thanks for your contribution! We should definitely support the gz compressed dumps.  I think this makes the `compressed` attribute less useful  (i.e. we should just check the compression type and store that).  We also need to make similar changes in the `_write_chunk` and `create_chunks` methods.  I'll add some specific comments to the PR.  If you are busy, I am happy to make all the changes in a separate PR.

File: qwikidata/json_dump.py [hunk 1]
```diff
@@ -16,7 +17,7 @@ class WikidataJsonDump:
     File names are of the form "wikidata-YYYYMMDD-all.json.bz2".  The file is a single JSON
```
Comment (galtay) [inline qwikidata/json_dump.py]: of the form "wikidata-YYYYMMDD-all.json[.bz2|.gz]"

File: qwikidata/json_dump.py [hunk 2]
```diff
@@ -30,13 +31,13 @@ def __init__(self, filename: str) -> None:
 
         if filename.endswith(".json"):
             self.basename, _ = os.path.splitext(filename)
-        elif filename.endswith(".json.bz2"):
+        elif filename.endswith((".json.bz2", "json.gz")):
```
Comment (galtay) [inline qwikidata/json_dump.py]: can we include the "." in front of "json.gz" for consistency with the other checks?

File: qwikidata/json_dump.py [hunk 3]
```diff
@@ -30,13 +31,13 @@ def __init__(self, filename: str) -> None:
...
             self.basename, _ = os.path.splitext(os.path.splitext(filename)[0])
         else:
-            raise ValueError('filename must end with ".json.bz2" or ".json"')
+            raise ValueError('filename must end with ".json.bz2" or "json.gz" or ".json"')
 
         self.filename = filename
-        self.compressed = filename.endswith("json.bz2")
+        self.compressed = filename.endswith((".json.bz2", "json.gz"))
```
Comment (galtay) [inline qwikidata/json_dump.py]: I think it makes more sense to have this variable be called something like `compression` and have its value set in the same if/else clause as `basename` with it getting one of the following values ["bz2", "gz", `None`]

File: qwikidata/json_dump.py [hunk 4]
```diff
@@ -47,8 +48,12 @@ def _open_dump_file(self) -> Iterator[IO[Any]]:
...
         """
         if self.compressed:
-            with bz2.open(self.filename, mode="rb") as fp:
-                yield fp
+            if self.filename.endswith(".json.bz2"):
+                with bz2.open(self.filename, mode="rb") as fp:
+                    yield fp
+            elif self.filename.endswith(".json.gz"):
+                with gzip.open(self.filename, mode="rb") as fp:
+                    yield fp
```
Comment (galtay) [inline qwikidata/json_dump.py]: I think we can flatten this out now, something like, 
```
if self.compression == "bz2":
...
elif self.compression == "gz":
...
else
...
```

**Review by galtay — CHANGES_REQUESTED** [review]

**Comment by galtay** [comment]
Thanks for the updates.  I will make the changes to the file chunking logic and then create a new release for Pypi (0.1.0).

**Review by galtay — APPROVED** [review]

**Comment by SultanOrazbayev** [comment]
@galtay Resolved your comments, except for the ```create_chunks``` section (not sure what's the best way to handle this, as it introduces an extra degree of complexity... should be OK, since the code can read '.bz2', so it's internally consistent at least).

**Pull request merged**